### PR TITLE
#23603 : Allow users to write code when creating a file - PR2

### LINKED ISSUE #23603: EC: Allow users to write code when create a file
### Is your feature request related to a problem? Please describe.
As a user, I want to be able to create code file easily.

VTL files are widely used in dotCMS and the only way to create one is created one with a code editor and then upload it.

### Describe the solution you'd like
1. When create a file, add a new code field, so users can write code and save the file.

2. Keep the functionality of the file input field, so the user can upload files from the file system.

### Design
Use a code editor for the code. We'll have another issue for the final UI.

### Acceptance criteria
- Allow user to write code to create a file

- Allow user to browse files

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* Allow users to write code when creating a file. A new method was added to the Temp File API and the Temp File Endpoint to allow back-end users to "upsert" a temporary file.
* The Text Area field is displayed when a new File Asset is created. This way, Users can either upload a binary file, or start typing its contents in the Text Area field.
* The initial mechanism used to create the content of the File Asset is the same one used when we're editing an existing text File Asset.

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp [hunk 1]
```diff
@@ -72,31 +83,76 @@
...
+						console.log(this.responseText);
+					}
+				});
 
+				xhr.onload = function() {
+					const jsonData = JSON.parse(xhr.response);
+					tempFileId = jsonData.tempFiles[0].id;
+					const elements = document.getElementsByName("<%= field.getFieldContentlet() %>");
+					for (let i = 0; i < elements.length; i++) {
+						if (elements[i].tagName.toLowerCase() == "input") {
```
Comment (manuelrojas) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp:134]:
```suggestion
						if (elements[i].tagName.toLowerCase() === "input") {
```

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp [hunk 2]
```diff
@@ -72,31 +83,76 @@
...
+					if(this.readyState === 4) {
+						console.log(this.responseText);
+					}
+				});
 
+				xhr.onload = function() {
+					const jsonData = JSON.parse(xhr.response);
+					tempFileId = jsonData.tempFiles[0].id;
+					const elements = document.getElementsByName("<%= field.getFieldContentlet() %>");
+					for (let i = 0; i < elements.length; i++) {
```
Comment (manuelrojas) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp:133]:
```suggestion
					for (const element of elements) {
```

**Review by fabrizzio-dotCMS — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp [hunk 3]
```diff
@@ -72,31 +83,76 @@
...
-<%@page import="com.dotmarketing.util.UtilMethods"%>
+				xhr.open("PUT", "/api/v1/temp/id/" + tempFileId);
+				xhr.setRequestHeader("Content-Type", "application/json");
+				xhr.send(data);
+			}
+		} else {
+			FileAssetAjax.saveFileText(contentletInode.value, text, '<%=field.getVelocityVarName()%>', {
+				async: false,
+				callback: function (data) {
+					console.log("savedText");
```
Comment (manuelrojas) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp:148]: Do we need these logs?

```suggestion
					console.warn("savedText");
```

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp [hunk 4]
```diff
@@ -72,31 +83,76 @@
...
+				}
+				const data = JSON.stringify({
+					"fileName": fileName,
+					"fileContent": text
+				});
+				const xhr = new XMLHttpRequest();
+
+				xhr.addEventListener("readystatechange", function() {
+					if(this.readyState === 4) {
+						console.log(this.responseText);
```
Comment (manuelrojas) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp:125]: Do we need these logs?
```suggestion
						console.warn(this.responseText);
```

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp:134]: Done!

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp:133]: Done!

**Review by manuelrojas — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp (hunk 3 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp:148]: Removing code.

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp (hunk 4 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_file_asset_text_inc.jsp:125]: Removing code.

**Review by manuelrojas — APPROVED** [review]

**Pull request merged**